joern-cli/frontends/x2cpg/src/main/scala/io/joern/x2cpg/passes/frontend/XTypeHintCallLinker.scala

```diff
@@ -0,0 +1,44 @@
+package io.joern.x2cpg.passes.frontend
+
+import io.shiftleft.codepropertygraph.Cpg
+import io.shiftleft.codepropertygraph.generated.{EdgeTypes, PropertyNames}
+import io.shiftleft.codepropertygraph.generated.nodes.{Call, Method}
+import io.shiftleft.passes.CpgPass
+import io.shiftleft.semanticcpg.language._
+import overflowdb.traversal.Traversal
+
+/** Attempts to set the <code>methodFullName</code> and link to callees using the recovered type information from
+  * [[XTypeRecovery]]. Note that some methods may not be present as they could be external and have been dynamically
```

Comment:
What is the effect of the method not being present here? Are you saying that you handle that condition?

Reply:
TLDR: I don't handle it, not sure if it's within the scope of this pass or not, but it could be.

Essentially the `MethodStubPass` has already happened at this point, yet we may find new external methods. So no new external method stubs are linked under `callee`, but the information about this new call may be stored under `methodFullName`. Ideally, we would be able to run `MethodStubPass` after this, or be able to create new stubs on the fly here.

For the sake of tagging from `dynamicTypeHintFullName`, `methodFullName`, and `typeFullName`, it's fine. For the sake of using `call.callee`, it would be an issue.

Reply:
PS: Similar situation with type stubs.

`MethodStubPass` could perhaps have an `object` function where, given a `builder` and some method info, could register a new method in the `builder`.